test/nohardware/wrapper.jl

```diff
@@ -0,0 +1,117 @@
+  @test length(devinfolist) == numdevs == 0
+  
+  # FT_GetDeviceInfoDetail tests...
+  @test_throws FT_STATUS_ENUM FT_GetDeviceInfoDetail(0)
+
+  # FT_ListDevices tests...
+  numdevs2 = Ref{UInt32}()
+  retval = FT_ListDevices(numdevs2, Ref{UInt32}(), FT_LIST_NUMBER_ONLY)
+  @test retval == nothing
+  @test numdevs2[] == numdevs
+  devidx = Ref{UInt32}(0)
+  buffer = pointer(Vector{Cchar}(undef, 64))
+  @test_throws ErrorException FT_ListDevices(devidx, buffer, FT_LIST_BY_INDEX|FT_OPEN_BY_SERIAL_NUMBER)
+
+  # FT_Open tests...
+  @test_throws FT_STATUS_ENUM FT_Open(0)
+
+  # FT_OpenEx tests...
+  # by description
+  @test_throws FT_STATUS_ENUM FT_OpenEx("", FT_OPEN_BY_DESCRIPTION)
+  # by serialnumber
+  @test_throws FT_STATUS_ENUM FT_OpenEx("", FT_OPEN_BY_SERIAL_NUMBER)
+  
+  # FT_Close tests...
+  handle = FT_HANDLE() # create with invalid handle...
+  @test_throws FT_INVALID_HANDLE FT_Close(handle)
+
+  # FT_Read tests...
+  buffer = zeros(UInt8, 5)
+  @test_throws FT_INVALID_HANDLE FT_Read(handle, buffer, 0) # read 0 bytes
+  @test buffer == zeros(UInt8, 5)
+  @test_throws AssertionError FT_Read(handle, buffer, 6) # read 5 bytes
+  @test_throws AssertionError FT_Read(handle, buffer, -1) # read -1 bytes
+  
+  # FT_Write tests...
+  buffer = ones(UInt8, 5)
+  @test_throws FT_INVALID_HANDLE FT_Write(handle, buffer, 0) # write 0 bytes
+  @test buffer == ones(UInt8, 5)
+  @test_throws AssertionError FT_Write(handle, buffer, 6) # write 6 bytes
+  @test_throws AssertionError FT_Write(handle, buffer, -1) # write -1 bytes
+  
+  # FT_SetDataCharacteristics tests...
+  @test_throws FT_INVALID_HANDLE FT_SetDataCharacteristics(handle, FT_BITS_8, FT_STOP_BITS_1, FT_PARITY_NONE)
+  # Bad values
+  @test_throws AssertionError FT_SetDataCharacteristics(handle, ~(FT_BITS_8 | FT_BITS_7), FT_STOP_BITS_1, FT_PARITY_NONE)
+  @test_throws AssertionError FT_SetDataCharacteristics(handle, FT_BITS_8, ~(FT_STOP_BITS_1 | FT_STOP_BITS_2), FT_PARITY_NONE)
+  @test_throws AssertionError FT_SetDataCharacteristics(handle, FT_BITS_8, FT_STOP_BITS_1, ~(FT_PARITY_NONE | FT_PARITY_EVEN))
+  # closed handle
+  
+  # FT_SetTimeouts tests...
+  timeout_read, timeout_wr = 200, 100 # milliseconds
+  @test_throws FT_INVALID_HANDLE FT_SetTimeouts(handle, timeout_read, timeout_wr)
+  @test_throws InexactError FT_SetTimeouts(handle, timeout_read, -1)
+  @test_throws InexactError FT_SetTimeouts(handle, -1, timeout_wr)
+  
+  # FT_GetModemStatus tests
+  @test_throws FT_INVALID_HANDLE FT_GetModemStatus(handle)
+  
+  # FT_GetQueueStatus tests
+  @test_throws FT_INVALID_HANDLE FT_GetQueueStatus(handle)
+  
+  # FT_GetDeviceInfo tests
+  @test_throws FT_INVALID_HANDLE FT_GetDeviceInfo(handle)
+
+  # FT_GetDriverVersion tests
+  @test_throws FT_INVALID_HANDLE FT_GetDriverVersion(handle)
+
+  # FT_GetLibraryVersion tests
+  version = FT_GetLibraryVersion()
+  @test version isa DWORD
+  @test version > 0
+  @test (version >> 24) & 0xFF == 0x00 # 4th byte should be 0 according to docs
```

Comment:
Be careful - is this true on all platforms? I think I recall it returning different things on Linux/Mac, indeed the version is different across platforms

Reply:
The docs were not clear on this point. I'm tempted to leave it and then find out if this fails on Linux etc and fix at that point. Thoughts?